chat/message.py

```diff
@@ -100,6 +101,24 @@ def from_record(cls, record):
         obj.record = record
         return obj
 
+    @classmethod
+    def delete(cls, message_id) -> Record:
+        """
+        Soft-delete a message
+        - Mark message as deleted
+        - Update last_message and last_read_message
+        """
+        message = Message.fetch(message_id)
+        if message is None:
+            raise MessageNotFoundException()
+
+        if message.record['deleted']:
+            raise AlreadyDeletedException()
+
+        message.record['deleted'] = True
+        message.save()
+        return message
+
```

Comment:
Should the delete be instance method as `save`?

I am wondering where we use `save`..? Any clue.